datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -3552,6 +3564,61 @@ mod tests {
         );
     }
 
+    #[test]
+    fn simplify_expr_case_when_true() {
+        // CASE WHEN true THEN 1 ELSE x END --> 1
```

Comment:
can we please add some negative tests too -- like `CASE WHEN  a THEN 1 ELSE 2 END`?